Classes/NSObject+RZImport.m

```diff
@@ -332,15 +332,28 @@ - (void)rzi_importValuesFromDict:(NSDictionary *)dict withMappings:(NSDictionary
     BOOL canOverrideImports = [self respondsToSelector:@selector( rzi_shouldImportValue:forKey: )];
     
     NSSet *ignoredKeys = [[self class] rzi_cachedIgnoredKeys];
-    
-    [dict enumerateKeysAndObjectsUsingBlock:^(NSString *key, id value, BOOL *stop) {
-    
+    NSOrderedSet *orderedKeys = [[self class] rzi_cachedOrderedKeys];
+
+    NSArray *dictKeys = dict.allKeys;
+
+    if ( orderedKeys.count > 0 ) {
+        dictKeys = [dictKeys sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+            NSUInteger idx1 = [orderedKeys indexOfObject:obj1];
+            NSUInteger idx2 = [orderedKeys indexOfObject:obj2];
+
```

Comment:
However, `NSArray` is really just a `CFArray`, which uses `typedef signed long CFIndex;`